src/main/main.ts

```diff
@@ -31,6 +31,19 @@ ipcMain.on('ipc-example', async (event, arg) => {
   event.reply('ipc-example', msgTemplate('pong'));
 });
 
+ipcMain.on('folder-selection', async (event) => {
+  if (!mainWindow) {
+    console.error('folder-selection: no main window');
+    return;
+  }
+  const result = await dialog.showOpenDialog(mainWindow, {
```

Comment:
is `dialog` from electron the way to get to the OS native file browser?

Kind of just commenting so I remember to look this up later.

Reply:
Yeah, it talks to the OS you are on, and opens the native dialog from whichever operating system you are in.
This is happening in the main process and it's done by Node. 
You could open the same dialog from the browser by rendering an `<input type='file' />` which, when clicked, opens a dialog to select a file or folder. However, the browser is highly restricted to what it can read/write. Whereas in the `main` process, you have a bit more freedom, since it's the user who installed the app, not the user just looking at a website.
Also, in the main process I think we have more processing power to process the images (I think), and it's not blocking the rendering process.